src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -198,14 +198,16 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
                 LocalDataStoreInstance.LocalDataStore = keyStore;
             }
 
-            services = new DefaultServiceProvider();
+            grainFactory = new GrainFactory();
+            IDictionary<Type,object> singletons = new Dictionary<Type, object> { { typeof(IGrainFactory), grainFactory } };
+            services = new DefaultServiceProvider(singletons);
```

Comment:
Sorry haven't been following lately but I thought you guys made it possible to plug-in any DI container?
It looks like you created your own container? :)